private_gpt/ui/ui.py

```diff
@@ -32,6 +42,23 @@ def __init__(self) -> None:
         # Cache the UI blocks
         self._ui_block = None
 
+    @staticmethod
+    def _curate_sources(sources: list[Chunk]) -> set[Source]:
+        return {
+            Source(
+                file=chunk.document.doc_metadata["file_name"]
+                if chunk.document.doc_metadata
+                and "file_name" in chunk.document.doc_metadata
+                else "-",
+                page=chunk.document.doc_metadata["page_label"]
+                if chunk.document.doc_metadata
+                and "page_label" in chunk.document.doc_metadata
+                else "-",
+                text=chunk.text,
+            )
+            for chunk in sources
+        }
```

Comment:
Could have been in `Source` class

```python
# Usage would look like
Source.curated_from_chunks(sources)
```

Reply:
There are two identical check made on `bool(chunk.document.doc_metadata)` - could be reduced to one

Additionally, accessing to `page_label` and `file_name` can still cause key errors (the calls to the dict should be made with `.get(key, _default_value)`

Reply:
Moved and rewrote the whole method addressing these two comments. Thanks!